src/components/legend/style.js

```diff
@@ -481,6 +482,124 @@ module.exports = function style(s, gd) {
             stylePie(pts, d0Mod, tMod);
         }
     }
+
+    function styleSpatial(d) { // i.e. maninly traces having z and colorscale
+        var trace = d[0].trace;
+
+        var useGradient;
+        var ptsData = [];
+        if(trace.visible) {
+            switch(trace.type) {
+                case 'histogram2d' :
+                case 'heatmap' :
+                    ptsData = [
+                        ['M-15,-2V4H15V-2Z'] // similar to contour
```

Comment:
Ha you went ahead and made the heatmap legend items span rectangles:

![image](https://user-images.githubusercontent.com/6675409/69646437-ce610f80-1035-11ea-9195-4d02b16f01cf.png)

instead of a square. I like it, but it might be nice to get a few +1s from other folks.